nntrainer/layers/activation_layer.cpp

```diff
@@ -50,16 +50,20 @@ void ActivationLayer::forwarding(sharedConstTensors in) {
   Tensor &hidden_ = net_hidden[0]->var;
   /// @note @a _act_fn is expected to work out of place and not modify @a input
   _act_fn(net_input[0]->var, hidden_);
+  if (activation_type == ActivationType::ACT_SOFTMAX)
+    tempTensor = hidden_.clone();
 }
 
 void ActivationLayer::calcDerivative(sharedConstTensors derivative) {
-  Tensor &deriv = net_hidden[0]->grad;
-  Tensor &ret = net_input[0]->grad;
+  Tensor &deriv = net_hidden[0]->var;
+  Tensor &ret = net_input[0]->var;
 
-  if (activation_type == ActivationType::ACT_SOFTMAX)
-    ret = _act_prime_fn(net_hidden[0]->var, ret, deriv);
-  else
-    ret = _act_prime_fn(net_input[0]->var, ret, deriv);
+  if (activation_type == ActivationType::ACT_SOFTMAX) {
+    ret = _act_prime_fn(tempTensor, ret, deriv);
+  } else {
+    tempTensor = net_input[0]->var.clone();
```

Comment:
why is there a clone here than just passing `net_input[0]->var` to _act_prime_fn?

Reply:
As the activation layers perform elementwise operation, this will works.
Softmax is not element wise and hence needs output value stored.